### DISCUSSION TITLE
Unable to upload files (unable to store files) or (cannot use PASV on IPv6)

### DISCUSSION THREAD

**Discussion by Fashtas** [GetPublii/Publii#1859] [Help] [answered]
I've getting similar errors to one I see people get "Error. Application was able to connect with your server but was unable to store files" when testing the FTP connection. 
![image](https://github.com/user-attachments/assets/0628e4af-9040-4ef5-a5b9-d0397598d7df)

I've tried every single iteration of forwardslash and folder name I can think of for the directory (which seemed to be the suggestion generally) to no avail and couldn't see anything in logs for the test connection to check WHY it is failing.
As far as I can tell the directory is "/" since the FTP account is locked to that root folder of that site?

I did realize that clicking "Sync your website" actually tried to do that despite the test initially always failing and did discover an error in the deployment-process.log

"[Mon, 09 Dec 2024 01:39:43 GMT] [connection] > 'PASV'
[Mon, 09 Dec 2024 01:39:43 GMT] [connection] < '425 You cannot use PASV on IPv6 connections. Use EPSV instead.\r\n'
[Mon, 09 Dec 2024 01:39:43 GMT] [parser] < '425 You cannot use PASV on IPv6 connections. Use EPSV instead.\r\n'
[Mon, 09 Dec 2024 01:39:43 GMT] [parser] Response: code=425, buffer='You cannot use PASV on IPv6 connections. Use EPSV instead.'
[Mon, 09 Dec 2024 01:39:43 GMT] ERROR UPLOAD FILE: webpage.html "

 ATM I'm just using Filezilla to upload the output folder to the FTP server (didn't really do much with it, worked out of the gate with the default settings)

**Comment by dziudek** [comment]
Hi @Fashtas ,

Did you tried to use `App settings -> Experimental features -> Use alternative FTP library for deployment`?

**Comment by Fashtas** [comment]
I didn't know that menu even existed! Had to hunt a bit for it... Thanks!

The "Test Connection" now works but the same errors appear in the log if you Sync.

> [Mon, 09 Dec 2024 22:14:54 GMT] [parser] < '425 You cannot use PASV on IPv6 connections. Use EPSV instead.\r\n'

**Reply by dziudek** [reply, marked as answer]
@Fashtas - thank you for reporting this :) In fact I have found a bug in the code - your logs suggests that you are still using old FTP library, because it contains `[parser]` string in the logs - new library does not have it.

To solve this - please enable option `App settings -> Experimental features -> Auto beautify source code in WYSIWYG Editor` - I have found a bug in the code which causes that this option is used in the alternative FTP flag during deployment (test connection is correct). The restart can be required after this change.

Bug is fixed and will be deployed in v.0.46.3 before Christmas. Sorry for such confusion!

ref: https://github.com/GetPublii/Publii/commit/18b6bfc1811529d3e2b975185cd26b8edfe643b7

**Comment by Fashtas** [comment]
This works perfectly now (with Auto beautify enabled) 
Thanks very much for your hard work!
